src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -65,6 +64,12 @@ private protected override INameResolver CreateResolver(PowerFxConfig alternateC
         {
             // The RecalcEngineResolver allows access to the values from UpdateValue. 
             var resolver = new RecalcEngineResolver(this, alternateConfig ?? Config);
+
+            if (resolver is ISetGlobalSymbols setSymbols)

```

Comment:
Why an interface + setter instead of adding this as an arg to the constructor?

Reply:
Trying to reduce the impact of the breaking change